src/imagings/hooks/useRequestImaging.tsx

```diff
@@ -14,22 +14,24 @@ export interface ImagingRequest {
   type: string
 }
 
-async function requestImaging(request: ImagingRequest): Promise<void> {
-  const error = validateImagingRequest(request)
+function requestImagingWrapper(user: any) {
+  return async function requestImaging(request: ImagingRequest): Promise<void> {
+    const error = validateImagingRequest(request)
 
-  if (!isEmpty(error)) {
-    throw error
-  }
+    if (!isEmpty(error)) {
+      throw error
+    }
 
-  await ImagingRepository.save({
-    ...request,
-    requestedBy: 'test',
-    requestedOn: new Date(Date.now()).toISOString(),
-  } as Imaging)
+    await ImagingRepository.save({
+      ...request,
+      requestedBy: user?.fullName || '',
```

Comment:
Hi @riiniii 
Note the requirements  
- _"The requestedBy should be the currently logged in user.id at the time the imaging request is created"_
- _"The list of imaging requests should display the user.fullName based on the imaging request's requestedBy"_

So in other words we must store the user id (not full name as is currently in this code) then when displaying the imaging request, we display the full username. 

![image](https://user-images.githubusercontent.com/5468546/104345295-d82e2680-5506-11eb-9208-5cb55c62254e.png)

Reply:
Hi @blestab, thank you for the clarification. I should have asked for clarifications ^^'

I ended up passing requestedByFullName alongside requestedBy when saving a new imaging request, and showing the fullName in the table.

Reply:
Okay great! Thanks